include/boost/geometry/algorithms/line_interpolate.hpp

```diff
@@ -130,7 +131,7 @@ struct interpolate_range
                                p,
                                diff_distance);
                 Policy::apply(p, pointlike);
-                if (std::is_same<PointLike, point_t>::value)
+                if (std::is_same<typename tag<PointLike>::type, point_tag>::value)
```

Comment:
FYI, we have type traits for this now, i.e. `util::is_point<PointLike>::value` here:
https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/util/type_traits.hpp#L38

Some compilers will complain about this `constant condition` but it was here before.

Reply:
Thanks for the hint, I will update the PR tomorrow to use is_point.

With regard to the constant condition: This was last solved with the BOOST_GEOMETRY_CONDITION macro, should that be used here as well?
Alternatively, there is the BOOST_IF_CONSTEXPR macro from boost.config, which expands to "if" for <=c++14 and "if constexpr" for >c++17 (which was apparently available in the compiler that generated the warning: https://github.com/boostorg/geometry/issues/727)